### ISSUE TITLE
Hope to add rerank module

### ISSUE THREAD

**Issue by 111Moderato** [milvus-io/milvus#2620]
**Is your feature request related to a problem? Please describe.**
Our retrieval system needs to rerank to get more reliable results.

**Describe the solution you'd like**
Can you integrate the rerank module of faiss into milvus?

**Comment by JinHai-CN** [comment]
Yes, we are going to provide pre /.post process. In postprocess part, rerank results for PQ or SQ index will be provided.

**Comment by 111Moderato** [comment]
Would you like to consider adding K-Reciprocal as a rerank method？ @JinHai-CN
